app/nginx-1.11.10/src/event/modules/ngx_ff_module.c

```diff
@@ -129,6 +129,31 @@ static int inited;
         real_##func;                                        \
     })
 
+extern intptr_t    ngx_max_sockets;
+
+/*-
+ * Make sockfd assigned by the fstack plus the value of maximum kernel socket. 
+ *  so we can tell them apart according to different scopes.
+ * Solve the condominium ownership at Application Layer and obtain more freedom.
+ * fstack tried to do this by 'fd_reserve', unfortunately, it doesn't work well.
```

Comment:
> fstack tried to do this by 'fd_reserve', unfortunately, it doesn't work well.

What does it mean? Is there any bug of 'fd_reserve'?

Reply:
Maybe i have understood what you mean.
As a configurable variable, it is not always equal to the real limit number of kernel file descriptors.
Emm, anther way is that assign the value of `rlimit` to `fd_reserve` when F-Stack  initializes.

Reply:
Sorry that the granularity of the submission is too large,  I can then split it into smaller ones. But one fact is the function involved many points,  splitting up can offer “function unavailable”. Probably, it's not a good set.

About "fd_reserve" , you got me. 
Also "reserve" felt wrong to me, I tend to "offset" in the glue layer to ensure that fd does not conflict with the kernel.

A worker of the native nginx has only one flow of events driven by multiplexing. But running on fstack, it should have two, one for fstack and one for kernel (I call it "Another Event-Driven Stream", "AEDS" for short).
Fstack already has a prototype to handle "channel message", I have improved it, to handle events on kernel. 
Directive "kernel_network_stack" can bind server to AEDS.

In our case, nginx should communicate with local process,  e.g. redis, db; And handle packet from kernel.